phoenix-core/src/main/java/org/apache/phoenix/expression/Expression.java

```diff
@@ -88,4 +88,9 @@
      * @return
      */
     boolean requiresFinalEvaluation();
+    
+    /**
+     * @return the estimated size in bytes of the expression when serialized
+     */
+    int getEstimatedByteSize();
```

Comment:
The only place where this is used is in IndexMaintainer serialize, if we are moving to Protobufs then this change is not worth it, I will remove it.